CRM/Financial/BAO/FinancialType.php

```diff
@@ -455,9 +455,12 @@ public static function checkPermissionToEditFinancialType($financialTypeID) {
   public static function isACLFinancialTypeStatus() {
     if (!isset(\Civi::$statics[__CLASS__]['is_acl_enabled'])) {
       \Civi::$statics[__CLASS__]['is_acl_enabled'] = FALSE;
-      $contributeSettings = Civi::settings()->get('contribution_invoice_settings');
-      if (CRM_Utils_Array::value('acl_financial_type', $contributeSettings)) {
-        \Civi::$statics[__CLASS__]['is_acl_enabled'] = TRUE;
+      $realSetting = \Civi::$statics[__CLASS__]['is_acl_enabled'] = Civi::settings()->get('acl_financial_type');
+      if (!$realSetting) {
+        $contributeSettings = Civi::settings()->get('contribution_invoice_settings');
+        if (CRM_Utils_Array::value('acl_financial_type', $contributeSettings)) {
+          \Civi::$statics[__CLASS__]['is_acl_enabled'] = TRUE;
+        }
```

Comment:
umm 
```php
\Civi::$statics[__CLASS__]['is_acl_enabled'] = Civi::settings()->get('acl_financial_type') ?: 
CRM_Utils_Array::value('acl_financial_type', Civi::settings()->get('contribution_invoice_settings'), FALSE);
```

Reply:
I guess so but I kinda wanted to parse out the bit that should be deprecated / removed